primefaces/src/main/java/org/primefaces/util/FileUploadUtils.java

```diff
@@ -71,31 +73,18 @@ public static String requireValidFilename(String filename) {
         }
 
         if (isSystemWindows()) {
-            if (!filename.contains("\\\\")) {
-                int length = FilenameUtils.getPrefixLength(filename);
-                String prefix = LangUtils.substring(filename, 0, length);
-                String[] parts = prefix.split(Pattern.quote(File.separator));
-                for (String part : parts) {
-                    if (INVALID_FILENAME_PATTERN.matcher(part).find()) {
-                        throw new FacesException("Invalid filename: " + filename);
-                    }
-                }
-            }
-            else {
+            if (!INVALID_FILENAME_WINDOWS.matcher(filename).find()) {
                 throw new FacesException("Invalid filename: " + filename);
             }
         }
-
-        String name = FilenameUtils.getName(filename);
-        String extension = FilenameUtils.getExtension(filename);
-        if (name.isEmpty() && extension.isEmpty()) {
-            throw new FacesException("Filename can not be the empty string");
+        else if (INVALID_FILENAME_LINUX.matcher(filename).find()) {
+            throw new FacesException("Invalid filename: " + filename);
         }
 
-        return name;
+        return FilenameUtils.getName(filename);
     }
 
-    public static String requireValidFilePath(String filePath) throws ValidationException {
+    public static String requireValidFilePath(String filePath, boolean mustExist) {
```

Comment:
Should we used ValidatorException instead, sounds appropriate don't you think?

Reply:
ValidatorException requires `FacesMessages` not strings so I think FacesException is OK here?

Reply:
You can always do something like that:

```java
throw new ValidatorException(new FacesMessage(FacesMessage.SEVERITY_ERROR, "Validation Error", value + " is not a valid email;"));
```

Reply:
Switched to ValidatorException